src/script/script.h

```diff
@@ -429,6 +429,20 @@ class CScript : public CScriptBase
 
     SERIALIZE_METHODS(CScript, obj) { READWRITEAS(CScriptBase, obj); }
 
+    CScript& operator+=(const CScript& b)
+    {
+        reserve(size() + b.size());
+        insert(end(), b.begin(), b.end());
+        return *this;
+    }
+
+    friend CScript operator+(const CScript& a, const CScript& b)
```

Comment:
I don't really like this. Does this also enable `CScript{} + CScriptBase{}`? If yes, that sounds bad, because you likely don't want to append data without pushing it.

Even if it doesn't, is there any downside in having a named function for this? The existing `Cat` might even work for this?

Reply:
Indeed, just using `Cat` does work.
```diff
diff --git a/src/script/miniscript.h b/src/script/miniscript.h
index c67a3c4ea0..88fa4730cd 100644
--- a/src/script/miniscript.h
+++ b/src/script/miniscript.h
@@ -446,7 +446,7 @@ public:
                 case NodeType::RIPEMD160: return CScript() << OP_SIZE << 32 << OP_EQUALVERIFY << OP_RIPEMD160 << node.data << (verify ? OP_EQUALVERIFY : OP_EQUAL);
                 case NodeType::HASH256: return CScript() << OP_SIZE << 32 << OP_EQUALVERIFY << OP_HASH256 << node.data << (verify ? OP_EQUALVERIFY : OP_EQUAL);
                 case NodeType::HASH160: return CScript() << OP_SIZE << 32 << OP_EQUALVERIFY << OP_HASH160 << node.data << (verify ? OP_EQUALVERIFY : OP_EQUAL);
-                case NodeType::WRAP_A: return (CScript() << OP_TOALTSTACK) + std::move(subs[0]) + (CScript() << OP_FROMALTSTACK);
+                case NodeType::WRAP_A: return Cat(Cat((CScript() << OP_TOALTSTACK), std::move(subs[0])), (CScript() << OP_FROMALTSTACK));
                 case NodeType::WRAP_S: return (CScript() << OP_SWAP) + std::move(subs[0]);
                 case NodeType::WRAP_C: return std::move(subs[0]) + CScript() << (verify ? OP_CHECKSIGVERIFY : OP_CHECKSIG);
                 case NodeType::WRAP_D: return (CScript() << OP_DUP << OP_IF) + std::move(subs[0]) + (CScript() << OP_ENDIF);
```

I'll look into adding a `Cat()` which takes a variadic arguments, instead of nesting them and creating more vectors than needed.

Reply:
220536